### PR TITLE
Use rustfmt as a crate

### PR DESCRIPTION
**Pull request by zeenix**
Instead of launching it as binary.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -25,6 +25,7 @@ lalrpop-intern = "0.12.5"
 error-chain = "0.10.0"
 quote = "0.3.15"
 unicode-xid = "0.0.4"
+rustfmt = "*"
```
Comment (nikomatsakis) [inline Cargo.toml]: Why `= "*"` and not `0.9`?

File: src/lib.rs [hunk 2]
```diff
@@ -37,8 +38,14 @@ pub fn gobject_gen(input: TokenStream) -> TokenStream {
...
-            println!("{}", output);
+            let mut config: rustfmt::config::Config = Default::default();
+            config.set().error_on_line_overflow(false);
+            let summary = rustfmt::run(rustfmt::Input::Text(token_stream.as_str().into()),
+                                       & config);
+            if !summary.has_no_errors() {
+                println!("{:#?}", summary);
+                panic!("cannot generate gobjects")
+            }
             token_stream.parse().unwrap()
```
Comment (nikomatsakis) [inline src/lib.rs]: I'm a bit confused here -- we are running rustfmt, but are we using the output for anything?

File: Cargo.toml (hunk 1 above)
Comment (zeenix) [inline Cargo.toml]: Ah, forgot to update.

File: src/lib.rs (hunk 2 above)
Comment (zeenix) [inline src/lib.rs]: AFAICT rustfmt.run() modifies the passed string. The only output is a Summary struct and is only implements Debug (not Display) so it's not worth printing out unless there is an error.

Comment (zeenix) [inline src/lib.rs]: hmm.. I just printed the tokens_stream before and after the call and it seems I'm wrong. I'll dig more into the API to see what I'm expected to do here..

Comment (zeenix) [inline src/lib.rs]: Right! So I was supposed to use another function. Everything works now.

**Comment by zeenix** [comment]
@nikomatsakis ping, all good now to merge?

**Comment by nikomatsakis** [comment]
LGTM! =)

**Pull request merged**